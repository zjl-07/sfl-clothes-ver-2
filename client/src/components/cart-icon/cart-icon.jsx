import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { toggleCartHidden } from "redux/cart/cart.action";
import { selectCartItemCount } from "redux/cart/cart.selector";
import { CartIconContainer, CountIconContainer } from "./cart-icon.style";

const CartIcon = ({ dispatch, itemCount }) => {
  return (
    <CartIconContainer onClick={() => dispatch(toggleCartHidden())}>
      <i className="fa fa-shopping-cart" aria-hidden="true"></i>
      <CountIconContainer>{itemCount}</CountIconContainer>
    </CartIconContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemCount
});

export default connect(mapStateToProps)(CartIcon);
