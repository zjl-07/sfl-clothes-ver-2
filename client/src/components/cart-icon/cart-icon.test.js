import React from "react";
import { shallow } from "enzyme";
import CartIcon from "./cart-icon";
import { CartIconContainer } from "./cart-icon.style";
import { toggleCartHidden } from "redux/cart/cart.action";

describe("Cart Icon", () => {
  let wrapper;
  let mockDispatch;
  let mockItemCount = 0;

  beforeEach(() => {
    mockDispatch = jest.fn();

    const mockProps = {
      itemCount: mockItemCount,
      dispach: mockDispatch
    };

    wrapper = shallow(<CartIcon.WrappedComponent {...mockProps} />);
  });

  it("expect Cart Icon to match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
