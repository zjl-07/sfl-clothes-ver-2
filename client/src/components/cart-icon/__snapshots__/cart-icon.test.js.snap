// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon expect Cart Icon to match snapshot 1`] = `ShallowWrapper {}`;
